samples/drivers/counter_rtc/sample.yaml

```diff
@@ -0,0 +1,13 @@
+sample:
+  name: Counter RTC Driver Sample
```

Comment:
Why are you calling this a "counter_rtc" sample? Why not just "counter"?

Also need to add license/copyright header to new files and a sample README.rst.

Reply:
Sure, so I finally changed to counter/alarm as it specifically plays with alarm.
README added